app/components/chart/settings.js

```diff
@@ -129,24 +89,30 @@ var Settings = React.createClass({
         ref="footer" />
       </div>
       );
-    
   },
   renderChart: function() {
-    
-    return (
-      <SettingsChart
-        bgUnits={this.props.bgPrefs.bgUnits}
-        patientData={this.props.patientData}
-        ref="chart" />
+
+    const SettingsChart = ChartFactory.getChart(
+      this.props.patientData.grouped.pumpSettings[0].source,
```

Comment:
Perhaps `_.get()` here to protect against something missing in the path? I think this might explode for someone who has only uploaded CGM data and not pump data, for example.